src/dune/coq_stanza.ml

```diff
@@ -52,8 +52,15 @@ module Extract = struct
     }
 
   let ml_target_fnames t =
+    let prelude = Coq_module.Name.to_string (snd t.prelude) in
     List.concat_map t.extracted_modules ~f:(fun m ->
         let m = Module_name.to_string m in
+        let m =
+          if String.Caseless.equal prelude m then
```

Comment:
@ejgallego I need this strange hack because it seems like coq prefers `Foo.ml` over `foo.ml` _unless_ there's a corresponding `foo.v`. Not sure what's the better way to handle this

Reply:
Umm, this was introduced in https://github.com/coq/coq/commit/2be0079f2ed4b67eae474341a10b9f60dcf83c4f , indeed I wonder what the best way would be.

Reply:
Likely the cleanest would be to take into account the different extraction modes; I guess this is the main source of the problem right?

Reply:
Thanks for the reference. I suppose my hack makes sense then. What about the filenames of dependencies, are they ever taken into account?

What are the different extraction modes?

Reply:
I mean `Extraction` vs `Separate Extraction` etc... actually, how did you manage to generate `Foo.ml` in your first message?